java/server/src/org/openqa/selenium/grid/server/BaseServerOptions.java

```diff
@@ -96,4 +101,24 @@ public URI getExternalUri() {
   public boolean getAllowCORS() {
     return config.getBool("server", "allow-cors").orElse(false);
   }
+
+  public boolean isSecure() {
+    return config.getBool("server", "https").orElse(false);
+  }
+
+  public File getPrivateKey() {
+    String privateKey = config.get("server", "https-private-key").orElse(null);
+    if (privateKey != null) {
+      return new File(privateKey);
+    }
+    throw new ConfigException("you must provide a private key via --https-private-key when using --https");
```

Comment:
If you're using the `--https` flag, check that value before throwing an exception.